### DISCUSSION TITLE
A way to get `.shape` of a discriminated union schema member?

### DISCUSSION THREAD

**Discussion by elieven** [colinhacks/zod#3386] [Q&A]
Is there some way of doing something like:

```typescript
const bigConfigSchema = z.object({
	someProperty: z.string(),
	someSubsection: z.discriminatedUnion('something', [
		z.object({
			something: z.literal('a'),
			thing1: z.boolean(),
			thing2: z.number(),
		}),
		z.object({
			something: z.literal('b'),
			thinga: z.object({
				thingaa: z.string(),
				thingab: z.string(),
			}),
			thingb: z.number(),
		})
	])
})

const b_configSchema = bigConfigSchema.shape.someSubsection.???
```

It would be particularly useful in cases where you define things like config schemas and want to use parts of that full config schema in different components and the config schema might have versions and you have defined different versions and conversion functions between them but don't want to do the same for separate unique definitions of each part of the config schema as you would quickly lose your mind. So just having that subpart of the config be defined as `.shape` of that part of the full config would be great (and is something you can do in typescript) but so far as I am aware you cannot do that directly for a discriminated union member.

**Comment by gsong** [comment]
`bigConfigSchema.shape.someSubsection.options` will get you an array of the schemas: <https://zod.dev/?id=discriminated-unions>
